src/StackExchange.Redis/PhysicalConnection.cs

```diff
@@ -1579,8 +1579,14 @@ private void MatchResult(in RawResult result)
                     }
 
                     // invoke the handlers
-                    var channel = items[1].AsRedisChannel(ChannelPrefix, RedisChannel.PatternMode.Literal);
-                    Trace("MESSAGE: " + channel);
+                    RedisChannel channel;
+                    if (items[0].IsEqual(message)) {
```

Comment:
feels like we're testing this a lot; can we hoist this to a `bool` somewhere? maybe:

``` c#
if (items.Length >= 3 && IsSimpleMessage(items[0], out bool isSharded))
```

then just use `isSharded` in all the places? for example `channel = items[1].AsRedisChannel(ChannelPrefix, RedisChannel.PatternMode.Literal, isSharded);`